src/textual/css/_styles_builder.py

```diff
@@ -462,9 +463,18 @@ def border_value_error():
                 except ColorParseError:
                     border_value_error()
 
+            elif token_name == "scalar":
+                alpha_scalar = Scalar.parse(token.value)
+                if alpha_scalar.unit != Unit.PERCENT:
+                    self.error(name, token, "alpha must be given as a percentage.")
+                border_alpha = alpha_scalar.value / 100.0
+
             else:
                 border_value_error()
 
+        if border_alpha is not None:
+            border_color = border_color.with_alpha(border_alpha)
```

Comment:
I don't think you should set the alpha, but rather multiply it.

If a color has 80% alpha, and you set the alpha value to 50%, then the border color should have 40% alpha.

Reply:
As discussed in person, this was following what the `color` property does in CSS right now, so that it worked the same.

As a followup from that conversation, we're going to change how `color` works so that it does the multiplication too, and then this will be modified to work the same (it's likely that we'll have `Color` grow a `multiply_alpha` method, or perhaps even implement `__mul__` (I think I prefer the former over the latter).